### PR TITLE
General cleanup

### PR DESCRIPTION
**Pull request by Veratil**
* converts newlines to unix newlines since it was mixed
* convert includes to C++ \<cxx> style
* fix some warnings
* remove unused test files
* add all tests back to test.cpp (disable failing ones with comments)
* test5.gro updated to current output format

### CODE REVIEW

**Comment by Be-ing** [comment]
For the future, I would prefer all of those bullet points to be separate commits, but I'm not going to make you go back and redo your work just for a nicer Git history.

File: src/allegro.cpp [hunk 1]
```diff
@@ -9,19 +9,19 @@
...
 
-#include "assert.h"
-#include "stdlib.h"
-#include "stdio.h"
-#include "string.h"
-#include "memory.h"
-#include <iostream>
+#include <cassert>
+#include <cmath>
+//#include <cstdlib> -- commented out and nothing failed to compile, maybe not needed
```
Comment (Be-ing) [inline src/allegro.cpp:14]: If it works on CI, just delete it rather than commenting it out.

File: src/allegrord.cpp [hunk 2]
```diff
@@ -1,774 +1,775 @@
...
+        string real_string = field.substr(1, last - 1);
+        return atof(real_string.c_str());
+    } else {
+        return (double) parse_key(field);
+    }
+}
+
+
+// it is the responsibility of the caller to delete
+// the seq
```
Comment (Be-ing) [inline src/allegrord.cpp:70]: For the future, let's clean this up to use Doxygen style `///` comments.

**Comment by Veratil** [comment]
I got the error MSVC throws with clang, ` truncation from 'char *' to 'long' `, and a `reinterpret_cast` got it to go away, but it seems it's still thrown for MSVC.

File: src/allegrord.cpp (hunk 2 above)
Comment (Veratil) [reply src/allegrord.cpp:70]: I hadn't even looked at those yet! 😄

**Comment by Veratil** [comment]
> 
> 
> For the future, I would prefer all of those bullet points to be separate commits, but I'm not going to make you go back and redo your work just for a nicer Git history.

I can rework it with another PR, that's no problem. Just wanted to get some feedback on the current stuff.

**Comment by Be-ing** [comment]
By the way, could you devendor this library from LMMS? I'm sure packagers would appreciate that.

**Comment by Veratil** [comment]
> 
> 
> By the way, could you devendor this library from LMMS? I'm sure packagers would appreciate that. You may want to use a CMake ExternalProject as a fallback, for example: https://github.com/mixxxdj/mixxx/blob/main/CMakeLists.txt#L1894

It's on my todo list.

**Pull request closed without merging**